MapboxNavigation/RouteMapViewController.swift

```diff
@@ -252,47 +259,122 @@ extension RouteMapViewController: NavigationMapViewDelegate {
+                style.addSource(source)
+                streetsSources.append(source)
+            }
+            
+            if let mapboxSteetsSource = streetsSources.first, style.layer(withIdentifier: roadLabelLayerIdentifier) == nil {
+                let streetLabelLayer = MGLLineStyleLayer(identifier: roadLabelLayerIdentifier, source: mapboxSteetsSource)
+                streetLabelLayer.sourceLayerIdentifier = "road_label"
+                streetLabelLayer.lineOpacity = MGLStyleValue(rawValue: 1)
+                streetLabelLayer.lineWidth = MGLStyleValue(rawValue: 20)
+                streetLabelLayer.lineColor = MGLStyleValue(rawValue: .white)
+                style.insertLayer(streetLabelLayer, at: 0)
+            }
+            
+            let userPuck = mapView.convert(possibleClosestCoordinateToRoute, toPointTo: mapView)
+            let features = mapView.visibleFeatures(at: userPuck, styleLayerIdentifiers: Set([roadLabelLayerIdentifier]))
+            var smallestLabelDistance = Double.infinity
+            var currentName: String?
+            
+            for feature in features {
+                var allLines: [MGLPolyline] = []
+                
+                if let line = feature as? MGLPolylineFeature {
+                    allLines.append(line)
+                } else if let lines = feature as? MGLMultiPolylineFeature {
+                    allLines = lines.polylines
+                }
+                
+                for line in allLines {
+                    let featureCoordinates =  Array(UnsafeBufferPointer(start: line.coordinates, count: Int(line.pointCount)))
+                    let slicedLine = polyline(along: stepCoordinates, from: possibleClosestCoordinateToRoute)
+                    
+                    let lookAheadDistance:CLLocationDistance = 10
+                    guard let pointAheadFeature = coordinate(at: lookAheadDistance, fromStartOf: polyline(along: featureCoordinates, from: possibleClosestCoordinateToRoute)) else { continue }
+                    guard let pointAheadUser = coordinate(at: lookAheadDistance, fromStartOf: slicedLine) else { continue }
+                    guard let reversedPoint = coordinate(at: lookAheadDistance, fromStartOf: polyline(along: featureCoordinates.reversed(), from: possibleClosestCoordinateToRoute)) else { continue }
+                    
+                    let distanceBetweenPointsAhead = pointAheadFeature - pointAheadUser
+                    let distanceBetweenReversedPoint = reversedPoint - pointAheadUser
+                    let minDistanceBetweenPoints = min(distanceBetweenPointsAhead, distanceBetweenReversedPoint)
+                    
+                    if minDistanceBetweenPoints < smallestLabelDistance {
+                        smallestLabelDistance = minDistanceBetweenPoints
+                        
+                        var key = "name"
+                        if let languages = Locale.preferredLanguages.first,
```

Comment:
The text in the label should match the map, turn banner, or voice instructions, even if we happen to have a translated name in the vector tiles that is closer to the user’s preferred language. After all, it’s important to be able to match what’s on screen to real-world signage.

Since the Directions API always uses OpenStreetMap’s `name` tag, we should always use the `name` attribute for consistency with the turn banner and voice instructions.